src/ui/src/components/blueprints/BlueprintsAutogen.vue

```diff
@@ -26,22 +26,29 @@
 				<BlueprintsGenerationLoader />
 			</template>
 		</BlueprintsAutogenContents>
-		<template v-if="!isBusy">
-			<div class="BlueprintsAutogen__panel">
-				<p class="BlueprintsAutogen__error">
-					{{ errorMessage }}
-				</p>
-				<div class="BlueprintsAutogen__buttons">
-					<WdsButton variant="tertiary" @click="handleCancel">
-						Skip
+		<div class="BlueprintsAutogen__panel">
+			<p class="BlueprintsAutogen__error">
+				{{ errorMessage }}
+			</p>
+			<div class="BlueprintsAutogen__buttons">
+				<template v-if="isBusy">
+					<WdsButton variant="tertiary" @click="handleBack">
+						Back
 					</WdsButton>
-					<WdsButton variant="secondary" @click="handleAutogen">
-						<i class="material-symbols-outlined">bolt</i>
-						Autogenerate agent
-					</WdsButton>
-				</div>
+				</template>
+				<WdsButton variant="tertiary" @click="handleCancel">
+					Skip
+				</WdsButton>
+				<WdsButton
+					variant="secondary"
+					:disabled="isBusy"
+					@click="handleAutogen"
+				>
+					<i class="material-symbols-outlined">bolt</i>
+					{{ isBusy ? "Autogenerating..." : "Autogenerate agent" }}
```

Comment:
Could you try the `loading` props instead of switching the text? It uses a loader icon from WDS lib.

```suggestion
					:loading="isBusy"
					@click="handleAutogen"
				>
					<i class="material-symbols-outlined">bolt</i>
					Autogenerate agent
```

Reply:
![image](https://github.com/user-attachments/assets/43162d23-1055-4264-a44d-4144c835af83)